### ISSUE TITLE
aio: TOC jumps over items if you scroll too fast, provide cleaner transition

### ISSUE THREAD

**Issue by sjtrimble** [angular/angular#17220]
**NICE TO HAVE**

**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
TOC jumps over items if you scroll too fast on a page. There is no visual indicator that sections are being passed until the user stops scrolling.

**Expected behavior**
Provide smooth transition on TOC items as user scrolls through a page's sections.

**Minimal reproduction of the problem with instructions**

- Example: https://aio-staging.firebaseapp.com/guide/setup
- Scroll down fast and then stop

**Comment by gkalpak** [comment]
This is on purpose. There is currently a 300ms "sampling" on scroll events. I'll lower that.
A few more change detections will be triggered, but that shouldn't be too costly.

**Comment by gkalpak** [comment]
Fixed by #17249.
